File: backend/app/project_ids.py
"""Канонические идентификаторы проектов (строка, не user_id)."""

from __future__ import annotations

# ЖК Верба · 1 очередь строительства — основной объект платформы.
CANONICAL_DEFAULT_PROJECT_ID = "verba-phase-1"

# Исторический ключ файлового/dev-режима — мигрируем в канонический.
LEGACY_DEFAULT_PROJECT_IDS = frozenset({"default", "DEFAULT", ""})


def normalize_project_id(raw: str | None) -> str:
    s = (raw or "").strip()
    if not s or s in LEGACY_DEFAULT_PROJECT_IDS:
        return CANONICAL_DEFAULT_PROJECT_ID
    cleaned = "".join(
        ch
        if ch.isalnum() or ch in ("_", "-", ".") or ("\u0400" <= ch <= "\u04FF")
        else "_"
        for ch in s
    )
    return (cleaned[:128] or CANONICAL_DEFAULT_PROJECT_ID)

File: lib/projectIds.ts
/** Канонический project_id для ЖК Верба · 1 очередь (общий SoT в PostgreSQL). */
export const CANONICAL_DEFAULT_PROJECT_ID = "verba-phase-1";

const LEGACY_DEFAULTS = new Set(["", "default", "DEFAULT"]);

export function normalizeProjectId(raw: string | null | undefined): string {
  const s = String(raw ?? "").trim();
  if (!s || LEGACY_DEFAULTS.has(s)) return CANONICAL_DEFAULT_PROJECT_ID;
  const cleaned = s
    .replace(/[^a-zA-Z0-9\u0400-\u04FF_.-]/g, "_")
    .slice(0, 128);
  return cleaned.length > 0 ? cleaned : CANONICAL_DEFAULT_PROJECT_ID;
}
